File: frontend/src/services/roleService.js
import api from "./api";

export const getAllRoles = async () => {

    const response = await api.get("/roles");

    return response.data;
};

export const createRole = async (roleData) => {

    const response = await api.post(
        "/roles",
        roleData
    );

    return response.data;

};

export const updateRole = async (roleId, roleData) => {

    const response = await api.put(
        `/roles/${roleId}`,
        roleData
    );

    return response.data;

};

export const deleteRole = async (roleId) => {

    const response = await api.delete(`/roles/${roleId}`);

    return response.data;

};

File: backend/src/main/java/com/corporate/analytics/service/RoleService.java
package com.corporate.analytics.service;
import java.util.List;
import com.corporate.analytics.dto.request.CreateRoleRequest;
import com.corporate.analytics.dto.response.RoleResponseDTO;
import com.corporate.analytics.dto.request.UpdateRoleRequest;
public interface RoleService {

    List<RoleResponseDTO> getAllRoles();
    RoleResponseDTO createRole(CreateRoleRequest request);
    RoleResponseDTO getRoleById(Integer id);
    RoleResponseDTO updateRole(Integer id, UpdateRoleRequest request);
    void deleteRole(Integer id);
}